TaskManager.js

```diff
@@ -67,6 +94,32 @@ class TaskManager {
 
    editTask(taskId, title, description, deadline) {
 
+      if (typeof(deadline) !== 'undefined') {
+
+         const dateFormats = [
+               
+            'YYYY-MM-DD',
+            'D.M.YYYY',
+            'DD.MM.YYYY',
+         
+         ];
+      
+         const checkDateFormat = moment(deadline, dateFormats, true).isValid();
+      
+         const parseDate = moment(deadline, dateFormats, true);
+         const inputDate = parseDate.toDate();
+         const currentDate = new Date();
+      
+         const checkStatusDate = inputDate > currentDate;
+         const result = checkStatusDate && checkDateFormat
+
+         if (!result) {
+
+            console.log('Incorrect date.');
+            return 1;
+         }
+      }
+
```

Comment:
H-m-m-m-m, I saw this code before. Decompose into a function.

Reply:
You are good👍